### PR TITLE
Shared subgraph docs

### PR DESCRIPTION
**Pull request by indie**
Some more tutorial and help docs

### CODE REVIEW

**Comment by indie** [comment]
See also:  https://github.com/apache/incubator-mxnet/pull/12502

File: doc/sphinx/source/shared/index.rst [hunk 1]
```diff
@@ -0,0 +1,24 @@
+.. shared/index: 
+
+Subgraphs 
+=========
+
+As the toolbox for Deep Learning (DL) developers becomes more powerful and 
+better able to target workloads with both local and remote data, it becomes 
+increasingly important to be able to work in collaborative spaces like
```
Comment (aprocter) [inline doc/sphinx/source/shared/index.rst]: I don't really understand this sentence... not sure what you mean by "collaborative spaces".

File: doc/sphinx/source/shared/mxnet_tutorial.rst [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+nGraph Compiler stack as an `experimental backend`_. Our approach provides 
+**more opportunities** to start working with different kinds of graph 
+optimizations **than would be available to the MXNet framework alone**, for 
+reasons outlined in our `features`_ documentation.
+
+.. TODO : Link to latest on mxnet when/if they do this instead of linking to PR;
+   keep in mind this tutorial will still work  regardless of the merge status of 
+   the experimental backend if you already use the ngraph-mxnet Github repo 
+
+The Intel nGraph Compiler provides an in-memory abstraction layer for converting
```
Comment (aprocter) [inline doc/sphinx/source/shared/mxnet_tutorial.rst]: `in-memory` is probably not relevant here (I'd just cut it)

File: doc/sphinx/source/shared/mxnet_tutorial.rst [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+optimizations **than would be available to the MXNet framework alone**, for 
+reasons outlined in our `features`_ documentation.
+
+.. TODO : Link to latest on mxnet when/if they do this instead of linking to PR;
+   keep in mind this tutorial will still work  regardless of the merge status of 
+   the experimental backend if you already use the ngraph-mxnet Github repo 
+
+The Intel nGraph Compiler provides an in-memory abstraction layer for converting 
+the mathematical representation of a DL model into an optimized-for-execution 
+format that can be understood *by* and run *on* multiple hardware backends. A
```
Comment (aprocter) [inline doc/sphinx/source/shared/mxnet_tutorial.rst]: I might ditch the emphasis on `by` and `on`.

File: doc/sphinx/source/shared/mxnet_tutorial.rst [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+simplifies development by providing a common API for framework and hardware 
+developers to jointly optimize a Deep Learning model; deployment is simplified 
+by bringing highly-optimized CPU performance to a wide range of MXNet models, 
+especially in inference.
+
+Future releases of the nGraph Library will improve training support and add 
+other hardware backends currently under development, including Nvidia\* GPU, 
+Intel GPU, and custom silicon like the Intel® Nervana™ NNP. Future releases are 
+planned to further simplify installation with ``pip``, and to offer support for 
+the MS Windows\* operating system.
```
Comment (aprocter) [inline doc/sphinx/source/shared/mxnet_tutorial.rst]: MS -> Microsoft (or just say Windows?)

File: doc/sphinx/source/shared/mxnet_tutorial.rst [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+   keep in mind this tutorial will still work  regardless of the merge status of 
+   the experimental backend if you already use the ngraph-mxnet Github repo 
+
+The Intel nGraph Compiler provides an in-memory abstraction layer for converting 
+the mathematical representation of a DL model into an optimized-for-execution 
+format that can be understood *by* and run *on* multiple hardware backends. A 
+primary goal of this integration is to provide a seamless development and 
+deployment experience to Data Scientists and Machine Learning Engineers. It 
+simplifies development by providing a common API for framework and hardware 
+developers to jointly optimize a Deep Learning model; deployment is simplified
```
Comment (aprocter) [inline doc/sphinx/source/shared/mxnet_tutorial.rst]: `jointly` seems slightly off... kinda makes it sound like framework/hardware work is tightly coupled, which is not what we're trying to say

File: doc/sphinx/source/shared/mxnet_tutorial.rst [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+   python3 -m venv .venv
+   . .venv/bin/activate
+   cd python
+   pip install -e .
+   cd ../
+
+Now we're ready to use nGraph to run any model on a CPU backend. Building MXNet 
+with nGraph automatically enabled nGraph on your model scripts, and you 
+shouldn't need to do anything special. If you run into trouble, you can disable 
+nGraph by setting ``MXNET_SUBGRAPH_BACKEND=1``. If you do see trouble, please
```
Comment (aprocter) [inline doc/sphinx/source/shared/mxnet_tutorial.rst]: Is it really `1` to disable? @mbrookhart

File: doc/sphinx/source/shared/mxnet_tutorial.rst [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+shouldn't need to do anything special. If you run into trouble, you can disable 
+nGraph by setting ``MXNET_SUBGRAPH_BACKEND=1``. If you do see trouble, please 
+report it and we'll address it as soon as possible.
+
+Running Resnet-50-V2 Inference
+------------------------------
+
+To show a working example, we'll demonstrate how MXNet may be used to run 
+Resnet-50 Inference. For ease, we'll consider the standard MXNet Resnet-50-V2 
+model from the `gluon model zoo`_, and we'll test with latency at ``batch_size=1``.
```
Comment (aprocter) [inline doc/sphinx/source/shared/mxnet_tutorial.rst]: strike `latency at`

File: doc/sphinx/source/shared/mxnet_tutorial.rst [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+------------------------------
+
+To show a working example, we'll demonstrate how MXNet may be used to run 
+Resnet-50 Inference. For ease, we'll consider the standard MXNet Resnet-50-V2 
+model from the `gluon model zoo`_, and we'll test with latency at ``batch_size=1``. 
+Note that the nGraph-MXNet bridge supports static graphs only (dynamic graphs 
+are in the works); so for this example, we begin by converting the gluon model 
+into a static graph. Also note that any model with a saved checkpoint can be 
+considered a "static graph" in nGraph. For this example, we'll presume that the 
+model has been deemed "trained".
```
Comment (aprocter) [inline doc/sphinx/source/shared/mxnet_tutorial.rst]: change `has been deemed "trained"` to `is pre-trained` or maybe `has already been trained`

File: doc/sphinx/source/shared/mxnet_tutorial.rst [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+with nGraph automatically enabled nGraph on your model scripts, and you 
+shouldn't need to do anything special. If you run into trouble, you can disable 
+nGraph by setting ``MXNET_SUBGRAPH_BACKEND=1``. If you do see trouble, please 
+report it and we'll address it as soon as possible.
+
+Running Resnet-50-V2 Inference
+------------------------------
+
+To show a working example, we'll demonstrate how MXNet may be used to run 
+Resnet-50 Inference. For ease, we'll consider the standard MXNet Resnet-50-V2
```
Comment (aprocter) [inline doc/sphinx/source/shared/mxnet_tutorial.rst]: I _think_ it's properly styled `ResNet` with a capital N.

File: doc/sphinx/source/shared/mxnet_tutorial.rst [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+into a static graph. Also note that any model with a saved checkpoint can be 
+considered a "static graph" in nGraph. For this example, we'll presume that the 
+model has been deemed "trained".   
+
+.. literalinclude:: ../../../examples/subgraph_snippets/mxnet-gluon-example.py
+   :language: python
+   :lines: 17-32
+
+
+To load the model into nGraph, we simply bind the symbol into a Executor.
```
Comment (aprocter) [inline doc/sphinx/source/shared/mxnet_tutorial.rst]: `an Executor`

Comment (indie) [reply doc/sphinx/source/shared/mxnet_tutorial.rst]: Typo fixed.

File: doc/sphinx/source/shared/mxnet_tutorial.rst (hunk 9 above)
Comment (indie) [reply doc/sphinx/source/shared/mxnet_tutorial.rst]: I have seen it so many ways.. We can edit this here.

File: doc/sphinx/source/shared/mxnet_tutorial.rst (hunk 8 above)
Comment (indie) [reply doc/sphinx/source/shared/mxnet_tutorial.rst]: done

File: doc/sphinx/source/shared/mxnet_tutorial.rst (hunk 7 above)
Comment (indie) [reply doc/sphinx/source/shared/mxnet_tutorial.rst]: done

File: doc/sphinx/source/shared/mxnet_tutorial.rst (hunk 5 above)
Comment (indie) [reply doc/sphinx/source/shared/mxnet_tutorial.rst]: Agree and removed .. this phrase might have been a remnant of the blog when we were going to publish  that, which was focused more on the collaboration as a joint effort

File: doc/sphinx/source/shared/mxnet_tutorial.rst (hunk 3 above)
Comment (indie) [reply doc/sphinx/source/shared/mxnet_tutorial.rst]: done

File: doc/sphinx/source/shared/mxnet_tutorial.rst (hunk 4 above)
Comment (indie) [reply doc/sphinx/source/shared/mxnet_tutorial.rst]: done

File: doc/sphinx/source/shared/mxnet_tutorial.rst (hunk 2 above)
Comment (indie) [reply doc/sphinx/source/shared/mxnet_tutorial.rst]: Done

File: doc/sphinx/source/shared/index.rst (hunk 1 above)
Comment (indie) [reply doc/sphinx/source/shared/index.rst]: Another remnant phrase from the blog we never published, I think it can be cut here too

**Comment by indie** [comment]
> What are "Shared subgraphs" anyway?

@postrational I was hopeful this was explained by the index and subsequent MXNet example.  I also think the onnx example will need edited since @diyessi is updating our `BatchNorm` ops, which means the outputs listed in the example will likely change.  We're basically explaining how the nGraph function graph can begin to be parsed, and what a shared subgraph API can do.  Shared subgraphs are not unique to MXNet; they are not unique to TF or ONNX either.  Any framework that wants to open up a shared subgraph API can do so provided it does (these) minimal things to nGraph.  

Does that make sense?

File: doc/sphinx/source/shared/onnx_ssg_tutorial.rst [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+These outputs can now be used to start working with, for example, the  `broadcast shapes`_ op: 
+
+.. code-block:: python
+
+   import ngraph as ng
+   runtime = ng.runtime(backend_name='CPU')
+   print(runtime)
+   <Runtime: Backend='CPU'>
+   import numpy as np
+   input_node = ng.constant([1, 136, 28, 28])
```
Comment (postrational) [inline doc/sphinx/source/shared/onnx_ssg_tutorial.rst]: What are we planning to do here?

**Comment by indie** [comment]
Moving a doc reorg to a different branch given our new "happy path" of pip install and other changes.
